skaffold.yaml

```diff
@@ -49,19 +49,8 @@ build:
   local:
     useBuildkit: false
 deploy:
-  kubectl:
-    manifests:
-    - ./kubernetes-manifests/adservice.yaml
-    - ./kubernetes-manifests/cartservice.yaml
-    - ./kubernetes-manifests/checkoutservice.yaml
-    - ./kubernetes-manifests/currencyservice.yaml
-    - ./kubernetes-manifests/emailservice.yaml
-    - ./kubernetes-manifests/frontend.yaml
-    - ./kubernetes-manifests/paymentservice.yaml
-    - ./kubernetes-manifests/productcatalogservice.yaml
-    - ./kubernetes-manifests/recommendationservice.yaml
-    - ./kubernetes-manifests/redis.yaml
-    - ./kubernetes-manifests/shippingservice.yaml
+  kustomize:
+    paths: [kustomize]
```

Comment:
Nim and I discussed this and I said I'd look into the possibility, but I haven't had a chance to go over the implementation with him.  I think this is a positive change as it keeps the skaffold deployment in line with the one we use with kustomize.  This is important now that we are using kustomize modules to do more than just change some configs.  This seems like the best (only?) way to use skaffold for testing kustomize modules that add new deployments like this collector.  Thoughts?

Reply:
**Suggestion**:
Could we undo this `skaffold.yaml` change in this pull-request?
I think there's a larger discussion to be had here and should be done in a separate pull-request. I would also like to test the development workflow because merging this `skaffold` change.
Reason we to discuss further: We don't want `kustomize/` to contain development manifests since a lot of demos and tutorials use the `main` branch (e.g., [this](https://cloud.google.com/service-mesh/docs/by-example/canary-deployment)). Reminder: The `kustomize/base/` samples are updated every time a new release is created.

Reply:
I'm not sure I understand your comments here and I think this warrants further discussion.  I'm happy to pull this skaffold change out but only _because_ I've already used it to test the changes and I'm confident in them.

The core of the issue (as I see it) is that we need some way to `skaffold run` the same configurations we're putting in `kustomize/`.  This is especially true for modules like this one that include new deployments and not just config changes.  There might be a better solution than the one I have been using, but it is aligned with our current recommendation of using kustomize by commenting/uncommenting modules in `kustomize/kustomization.yaml`.  This should not interfere with any demos and tutorials since it does not change any of the manifest files and the default behavior will be to follow the default kustomize deployment (without any changes.)

We can remove these changes, but we'll need _some_ way of testing kustomize changes, and I think that maintaining parallel configs (one in kustomize, one in skaffold) for every combination of modules is not feasible.

Reply:
I 100% agree that we should `skaffold`-ify our development manifests.
And thank you so much for looking into this. I think the solution that you started is clever (e.g., having a new Kustomize Component for development).

But let's take this to another pull-request since it's somewhat unrelated to OpenTel, and because there are some details we need to flesh out such as:
1. could we use Kustomize's [`images`](https://github.com/NimJay/intro-to-kustomize/blob/main/images/kustomization.yaml) transformer or something similar?
1. how should we update the doc inside [`/docs/development-guide.md`](https://github.com/GoogleCloudPlatform/microservices-demo/blob/main/docs/development-guide.md)?
1. should we delete `/kubernetes-manifests/*`?
1. how should the release automation inside `hack/` be updated?
1. how will this effect people using the `main` branch in their demos/tutorials? Should we introduce a new `git` tag or branch `latest` that points to the latest release of this repository?

I would also love to flesh out some of these details via a Google Meet. 
What you think, Drew?